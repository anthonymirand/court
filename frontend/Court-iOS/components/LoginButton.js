import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'expo';

import { Platform, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import Colors from '../constants/Colors';


/**
* Button that initiates facebook user authentication flow
*/
export default class LoginButton extends React.Component {
  render() {
    const { onPress, showLogo, text } = this.props;

    const fbIcon = (
      <Icon.Ionicons
        name="logo-facebook"
        size={30}
        style={{ marginRight: 15 }}
        color={Colors.teal}
      />
    );

    return (
      <TouchableOpacity onPress={onPress} activeOpacity={0.5}>
        <View style={styles.loginButton}>
          {showLogo && fbIcon}
          <Text style={styles.loginText}>{ text }</Text>
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  loginButton: {
    ...Platform.select({
      ios: {
        shadowColor: 'black',
        shadowOffset: { height: 8 },
        shadowOpacity: 0.15,
        shadowRadius: 9,
      },
      android: {
        elevation: 20,
      },
    }),
    backgroundColor: '#fff',
    borderRadius: 10,
    padding: 15,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  loginText: {
    color: Colors.teal,
    fontSize: 20,
  }
});

LoginButton.propTypes = {
  /**
  * Function that handles touch event
  */
  onPress: PropTypes.func.isRequired,
  /**
  * Boolean indicating whether or not to show the facebook logo
  */
  showLogo: PropTypes.bool.isRequired,
  /**
  * Text to be rendered in button
  */
  text: PropTypes.string.isRequired,
}
