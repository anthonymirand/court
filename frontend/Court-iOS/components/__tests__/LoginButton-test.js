import 'react-native';
import React from 'react';
import LoginButton from '../LoginButton';
import renderer from 'react-test-renderer';

it('Renders Default LoginButton', () => {
  const onPress = jest.fn();
  const tree = renderer.create(<LoginButton onPress={onPress} showLogo={true} text="Continue with Facebook" />).toJSON();

  expect(tree).toMatchSnapshot();
});
