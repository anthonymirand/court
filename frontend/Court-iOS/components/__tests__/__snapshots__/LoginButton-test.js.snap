// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Default LoginButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 15,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 8,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 9,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#21ACA5",
          "fontSize": 20,
        }
      }
    >
      Continue with Facebook
    </Text>
  </View>
</View>
`;
